### PR TITLE
MINOR: cleanup warnings in Kafka Streams code base

### PR DESCRIPTION
**Pull request by mjsax**
Cleanup for packages `org.apache.kafka.streams.*` and `org.apache.kafka.streams.errors.*`.

### CODE REVIEW

File: streams/src/main/java/org/apache/kafka/streams/StreamsBuilder.java [hunk 1]
```diff
@@ -87,12 +87,12 @@ public StreamsBuilder() {
      */
     @SuppressWarnings("this-escape")
     public StreamsBuilder(final TopologyConfig topologyConfigs) {
-        topology = getNewTopology(topologyConfigs);
+        topology = newTopology(topologyConfigs);
         internalTopologyBuilder = topology.internalTopologyBuilder;
         internalStreamsBuilder = new InternalStreamsBuilder(internalTopologyBuilder);
     }
 
-    protected Topology getNewTopology(final TopologyConfig topologyConfigs) {
```
Comment (guozhangwang) [inline streams/src/main/java/org/apache/kafka/streams/StreamsBuilder.java:95]: Have we double check that there's no callers outside KS code (like in other repos) of this function? Just double checking.

**Review by guozhangwang — APPROVED** [review]
One concern is that if other repos (I'm thinking ksql) may ever call `getNewTopology`, and if not please feel free to merge.

File: streams/src/main/java/org/apache/kafka/streams/StreamsBuilder.java (hunk 1 above)
Comment (mjsax) [reply streams/src/main/java/org/apache/kafka/streams/StreamsBuilder.java:95]: I could not find anything. Maybe @ableegoldman can comment? -- Also no big deal to leave this one. In the end, we plan to remove this experimental feature in the future anyway.

Comment (ableegoldman) [reply streams/src/main/java/org/apache/kafka/streams/StreamsBuilder.java:95]: I mean...there are definitely some people using named topologies, though I've made sure they are aware it is not considered a public API and not intended for public use without a KIP

Anyways given that this specific API is not even a public method I think it's fair game to remove.

Comment (mjsax) [reply streams/src/main/java/org/apache/kafka/streams/StreamsBuilder.java:95]: This PR does not remove it, we just change the name, so code could break. But yes, it's not public API so it should be ok. Going to merge. If there are concerns raised later, we can always add a commit to revert before 3.7 release.

**Pull request merged**